rust_src/src/symbols.rs

```diff
@@ -154,6 +155,18 @@ impl LispSymbolRef {
     }
 }
 
+impl LispStructuralEqual for LispSymbolRef {
+    fn equal(
+        &self,
+        _other: Self,
+        _equal_kind: equal_kind::Type,
+        _depth: i32,
+        _ht: &mut LispHashTableRef,
+    ) -> bool {
+        false
```

Comment:
Can't a symbol equal itself? (`s.equal(s)`)

Reply:
Yes, it should. This is left over from an earlier iteration. Thanks for noticing.

In the C code, the `EQ(o1, o2)` handles this.